cpp/src/gandiva/filter.h

```diff
@@ -45,6 +45,10 @@ class FilterCacheKey {
 
   bool operator!=(const FilterCacheKey& other) const { return !(*this == other); }
 
+  bool operator<(const FilterCacheKey& other) const {
+    return uniqifier_ < other.uniqifier_;
```

Comment:
This is not correct I believe. Also why is this needed?

Reply:
We needed to implement it because actually, we use this Filter key in a pair structure inside a set on the new Gandiva Cache.
We use this set to keep registers ordered by the elapsed build time.
This is why we need to implement this operator anyway, otherwise we can not use it as I needed in logic.

Reply:
You should compare on all the properties. uniqifier property is most likely 0 for all expressions

Reply:
Not sure if I completely understand your point, but I compared all possible properties for the bool operator<